### ISSUE TITLE
Bot name is very incovinient to access from UpdateParser

### ISSUE THREAD

**Issue by maurges** [fizruk/telegram-bot-simple#52]
I found today it's really hard to parse commands in UpdateParser, becaus I need to parse the `@my_bot_name` part, but I don't have access to the name.

As far as I understand, I can get bot's User from BotM via `reader botContextUser`, or from BotEnv which is internal.

I can't use BotM's reader because that is created after botAction is created.

I can sort of use BotEnv, but it comes with two big downsides: I need to replicate `startBot_` myself, and I need to use recursive do because, again, botAction must be created before I run `startBotEnv`

**Comment by swamp-agr** [comment]
I think the easiest way is to define your bot settings outside before bot start.
Then,
- either pass `Settings` to your Model handlers as an extra argument. 
- or use `Reader` to wrap them up in the application context.
- See `GameBot` for more details: https://github.com/fizruk/telegram-bot-simple/blob/master/examples/GameBot.hs

The thing is, in a chat your bot have to distinguish `@my_bot_name` from `@yet_another_bot_name` or from `@username`. You might also want to handle these as well based on the use-case.

**Comment by swamp-agr** [comment]
- Fixed via #121.
- Helper `commandWithBotName` was added.
- Please find release on Hackage: https://hackage.haskell.org/package/telegram-bot-simple-0.5.2
